providers/implementations/ciphers/cipher_chacha20.c

```diff
@@ -64,6 +65,19 @@ static void chacha20_freectx(void *vctx)
     }
 }
 
+static void *chacha20_dupctx(void *vctx)
+{
+    PROV_CHACHA20_CTX *ctx = (PROV_CHACHA20_CTX *)vctx;
+    PROV_CHACHA20_CTX *dupctx = NULL;
+    if (ctx != NULL) {
+        dupctx = OPENSSL_memdup(ctx, sizeof(*dupctx));
+        if (dupctx != NULL && dupctx->base.tlsmac != NULL) {
+            dupctx->base.tlsmac = OPENSSL_memdup(dupctx->base.tlsmac, dupctx->base.tlsmacsize);        
```

Comment:
This should be checked for failure and the dupctx needs to be freeed in such case.

Reply:
fixed